pkg/annotation/annotation.go

```
@@ -0,0 +1,199 @@
+	IgnoreBlocks []*IgnoreBlock
+}
+
+// IgnoreBlock represents a single block of ignore profiling data.
+type IgnoreBlock struct {
+	Annotation string   // concrete ignore pattern
+	Contents   []string // ignore contents
+	Lines      []int    // corresponding code line number of the ignore contents
+}
+
+// ParseIgnoreProfiles parses ignore profile data in the specified file and returns a ignore profile.
+func ParseIgnoreProfiles(fileName string) (*IgnoreProfile, error) {
+	pf, err := os.Open(fileName)
+	if err != nil {
+		return nil, err
+	}
+	defer pf.Close()
+
+	profile, err := parseIgnoreProfilesFromReader(pf)
+	profile.Filename = fileName
+	return profile, err
+}
+
+// parseIgnoreProfilesFromReader parses ignore profile data from the Reader and returns a ignore profile.
+func parseIgnoreProfilesFromReader(rd io.Reader) (*IgnoreProfile, error) {
+	s := bufio.NewScanner(rd)
+	lineNo := 0
+
+	profile := &IgnoreProfile{
+		Lines: make(map[int]bool),
+		Type:  BLOCK_IGNORE,
+	}
+
+	for s.Scan() {
+		lineNo++
+		line := s.Text()
+		match := IgnoreRegexp.FindStringSubmatch(line)
+
+		// not match, continue next line
+		if match == nil {
+			continue
+		}
+		if len(match) < 2 {
+			continue
+		}
+
+		// match contains the result of the regexp on IgnoreRegexp
+		// match = ["//+gocover:ignore:all", "all"] when input is `//+gocover:ignore:all`,
+		// match = ["//+gocover:ignore:block", "block"] when input is `//+gocover:ignore:block`,
+		// match = ["//+gocover:ignore:10", "10"] when input is `//+gocover:ignore:10`,
+		ignoreKind := match[1]
+		if ignoreKind == "all" { // set type to ALL_IGNORE and skip further processing
+			profile.Type = ALL_IGNORE
+			break
+		} else if ignoreKind == "block" { // block
+			skipLineCnt := ignoreOnBlock(s, profile, lineNo, line)
+			lineNo += skipLineCnt
+		} else { // number
+			// parse number fail means it's not the supported pattern, continue next line
+			total, err := strconv.Atoi(ignoreKind)
+			if err != nil {
+				continue
```

if some one type in 

//+gocover:ignore:fufufu

I think we should fail fast intead of ingoring the error.